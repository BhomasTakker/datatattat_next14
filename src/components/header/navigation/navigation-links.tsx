import React from "react";
import { NavLink, NavLinkData } from "./nav-link";
import styles from "./navigation-links.module.scss";

type NavigationProps = {
	navLinks: NavLinkData[];
	edit: boolean;
};

export const NavigationLinks = ({ navLinks, edit }: NavigationProps) => {
	const drawNavLinks = (links: NavLinkData[]) => {
		return links.map((link) => {
			const element = (
				<NavLink
					route={link.route}
					label={link.label}
					key={link.label}
					isEdit={edit}
				></NavLink>
			);
			return element;
		});
	};

	// Double div so we can center the nav
	return <div className={styles.root}>{drawNavLinks(navLinks)}</div>;
};
