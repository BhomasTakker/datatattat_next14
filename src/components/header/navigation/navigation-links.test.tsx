import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { NavigationLinks } from "./navigation-links";

jest.mock("./nav-link", () => {
	return {
		// __esModule: true,
		NavLink: ({ route, label }: { route: string; label: string }) => {
			return (
				<a data-testid={label} href={route}>
					{label}
				</a>
			);
		},
	};
});

const MOCK = [
	{
		route: "/home",
		label: "Home",
	},
	{
		route: "/about",
		label: "About",
	},
];

describe("Navigation Links", () => {
	it("renders provided links", () => {
		render(<NavigationLinks navLinks={MOCK} />);
		const link = screen.getByRole("link", { name: /home/i });
		expect(link).toBeInTheDocument();
	});

	it("renders a link with the provided href", () => {
		render(<NavigationLinks navLinks={MOCK} />);
		const links = screen.getAllByRole("link");
		expect(links.length).toBe(MOCK.length);
	});

	it("renders navigation unchanged", () => {
		const { container } = render(<NavigationLinks navLinks={MOCK} />);
		expect(container).toMatchSnapshot();
	});
});
