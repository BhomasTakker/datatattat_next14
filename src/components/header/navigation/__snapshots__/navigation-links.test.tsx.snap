// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Navigation Links renders navigation unchanged 1`] = `
<div>
  <div
    class="root"
  >
    <a
      data-testid="Home"
      href="/home"
    >
      Home
    </a>
    <a
      data-testid="About"
      href="/about"
    >
      About
    </a>
  </div>
</div>
`;
